webrtc.html

```diff
@@ -4790,23 +4770,74 @@ <h2>Methods</h2>
+                      the ECMAScript object represented by <var>keygenAlgorithm</var> to an
+                      <code>RTCCertificateExpiration</code> dictionary.</li>
+                    <li>If the conversion fails with an <var>error</var>,
+                      return a promise that is <a>rejected</a> with this <var>error</var>.</li>
+                    <li>If <var>certificateExpiration</var>.<code>expires</code>
+                      is not undefined, set <var>expires</var> to
+                      <var>certificateExpirations</var>.<code>expires</code>.</li>
+                  </ol>
+                </li>
+                <li>Let <var>normalizedKeygenAlgorithm</var> be the result of
+                  <a data-cite="!WebCryptoAPI#dfn-normalize-an-algorithm">normalizing an algorithm</a>
+                  with an operation name of <code>generateKey</code> and a
+                  <a data-cite="!WebCryptoAPI#dfn-supportedAlgorithms">supportedAlgorithms</a>
+                  value specific to production of certificates for
+                  <code>RTCPeerConnection</code>.</li>
+                <li>If the above step produces an <var>error</var>, return a
+                  promise that is <a>rejected</a> with this
+                  <var>error</var>.</li>
+                <li>If the <var>normalizedKeygenAlgorithm</var> parameter
+                  identifies an algorithm that the <a>user agent</a> cannot
+                  or will not use to generate a certificate for
+                  <code>RTCPeerConnection</code>, return a promise that is
+                  <a>rejected</a> with a <code>DOMException</code> of type
+                  <code>NotSupportedError</code>. In particular,
+                  <var>normalizedKeygenAlgorithm</var> MUST be an asymmetric
+                  algorithm that can be used to produce a signature used to
+                  authenticate DTLS connections.</li>
+                <li>Let <var>p</var> be a new promise.
+                <li>Run the following steps in parallel:
+                  <ol>
+                      <li>Perform the generate key operation specified by
+                        <var>normalizedKeygenAlgorithm</var> using
+                        <var>keygenAlgorithm</var>.</li>
+                      <li>Let <var>generatedKeyingMaterial</var> and
+                        <var>generatedKeyCertificate</var> be the private
+                        keying material and certificate generated by the
+                        above step.</li>
+                      <li>Let <var>certificate</var> be a new
+                        <code><a>RTCCertificate</a></code> object.</li>
+                      <li>Set <var>certificate</var>.[[\Expires]] to the
+                        current time plus <var>expires</var> value. A <a>user
+                        agent</a> MAY choose to limit the value of the
```

Comment:
I apologise for the sidetrack but what was the rationale for this limitation?

Reply:
I was not able to track when this was added, but it has been there since a lot of time.

Reply:
Perhaps @martinthomson knows? :slightly_smiling_face:

Reply:
Which limitation?

Reply:
That the user agent (which I read as *browser*) MAY choose to limit the *expires* value.

Reply:
Because browsers (or the users they act for) might not be well served by long-lived keys. And an expiration encourages/forces sites to deal with key changes. Maybe one days we will want to cleanly change requirements for keys and algorithms. Having keys expire prepares the ecosystem for that being something less than a massive disruption.

Reply:
Isn't the question related to why it is a MAY instead of a SHOULD or event a MUST?
It seems that Chrome, Firefox and Safari cap expires by a 1 year.
Maybe we should engrave that in the spec.

Reply:
I'd be OK with that, but I wouldn't want a hard requirement to support a maximum that long.  I would suggest making a separate issue for tracking that though.

Reply:
Making a suggestion for the range is something I would also appreciate. Otherwise, it's not really clear what's *acceptable*, not even for browser implementations. Also agree we should create a new issue and, once again, apologise for the sidetrack. :slightly_smiling_face:

Reply:
Filed https://github.com/w3c/webrtc-pc/issues/2137